DataFormats/NanoAOD/interface/FlatTable.h

```diff
@@ -26,7 +27,8 @@ namespace nanoaod {
       inline float one(const float &val) const {
         return (bits_ > 0 ? MiniFloatConverter::reduceMantissaToNbitsRounding(val, bits_) : val);
       }
-      inline void bulk(boost::sub_range<std::vector<float>> data) const {
+      template <typename Span>
+      inline void bulk(Span &&data) const {
```

Comment:
Why rvalue reference instead of
```suggestion
      inline void bulk(Span const& data) const {
```
?

Reply:
I once read this blog post about move semantics:
https://eli.thegreenplace.net/2014/perfect-forwarding-and-universal-references-in-c/#id6

Specifically the part "Don't let `T&&` fool you here ...". So it's not a rvalue reference, but makes the function automatically deduce if taking the argument by value or by reference is appropriate. Therefore it appeared to me that it might be a good habit to use `&&` by default for template deduced arguments, especially in this class where `bulk` is used with an rvalue.

Reply:
`T&&` is sometimes referred to as the _universal reference_.

Reply:
Maybe this isocpp blogpost is a more official reference for these universal references:
https://isocpp.org/blog/2012/11/universal-references-in-c11-scott-meyers

Reply:
Oh right, forgot that detail, never mind then.

Reply:
In this particular case I do not see a need for the universal reference as the type is directly interacted with instead of passed to another function via the use of `std::forward`.

Reply:
That's also true

Reply:
But it's such a minor change that I think I can also do it later in one go when I rebase this PR for the nanoAOD repository, or if you have more major comments, if this is okay.

Speaking about perfect forwarding, I recently noticed in the PR where I used FWCore/SOA that the SOA Table might make several unnecessary copies because even though `std::forward` is used, this is not matched by the corresponding universal reference for the argument, for example here (and also in other functions for the implementation details):
https://github.com/cms-sw/cmssw/blob/master/FWCore/SOA/interface/Table.h#L145

I didn't open an issue about this because I didn't think it was that important, but I it's just something to keep in mind for the next time anyone will work with FWCore/SOA.

Reply:
> But it's such a minor change that I think I can also do it later in one go when I rebase this PR for the nanoAOD repository, or if you have more major comments, if this is okay.

Perfectly fine for me, it is minor indeed.

> I didn't open an issue about this because I didn't think it was that important, but I it's just something to keep in mind for the next time anyone will work with FWCore/SOA.

The best way to remind us (`core`) is to open an issue :)

Reply:
Ok thanks!